internal/store/store.go

```diff
@@ -57,15 +60,33 @@ type BlockStore struct {
 	height int64
 }
 
+type BlockStoreOption func(*BlockStore)
+
+// WithMetrics sets the metrics.
+func WithMetrics(metrics *Metrics) BlockStoreOption {
```

Comment:
I like this way better than the `NewStore` function in StateStore. Maybe I'm too Rust-oriented.

Reply:
Not Rust oriented, we have this Builder pattern in several places in the codebase including the RPC APIs, I think it's a good pattern too and it helps with unit testing.